unidist/core/backends/mpi/core/shared_object_store.py

```diff
@@ -431,18 +431,17 @@ def _write_to_shared_buffer(self, data_id, reservation_data, serialized_data):
         for i, raw_buffer in enumerate(raw_buffers):
             raw_buffer_first_index = last_prev_index
             raw_buffer_len = len(raw_buffer)
-            raw_buffer_last_index = raw_buffer_first_index + len(raw_buffer)
-            if s_data_last_index > last_index:
+            last_prev_index = raw_buffer_first_index + len(raw_buffer)
```

Comment:
I guess this is not related to the issue this PR resolves. Discard the changes in this file.

Reply:
Why not? This isn't a big change, but it helps catch reserved memory overflows.

Reply:
If it helps to catch reserved memory overflows, how is this related to the get improvement? If it is not related, it is certainly a separate issue and should be addressed in a separate PR. We want to keep changes isolated in order for us to easily sort them out in future.

Reply:
It's just renaming a variable!
Every change included in the PR was made while working on the issue. Every task can be divided into its simplest subtask, but not every subtask needs to be a separated issue.

Reply:
I am disagree, but it is ok.